### PR TITLE
Actually return a non-zero error code on errors.

### PR DESCRIPTION
**Pull request by josepht**
LP:#1602394

### CODE REVIEW

File: snapcraft/internal/parser.py [hunk 1]
```diff
@@ -358,4 +359,4 @@ def _write_parts_list(path, master_parts_list):
 
 
 if __name__ == '__main__':  # pragma: no cover
-    main()                  # pragma: no cover
+    sys.exit(main())        # pragma: no cover
```
Comment (sergiusens) [inline snapcraft/internal/parser.py]: does dis mean the error code is 1 for one wiki error, 2 for two and so on? Is this what we want? Or do we want

```python
wiki_errrors = main()
if wiki_errors:
    sys.exit(1)
```

@elopio ?

Comment (josepht) [inline snapcraft/internal/parser.py]: Good catch.  You're right we want 1 or 0, maybe another in the future for things like network issues getting the index.

File: snapcraft/internal/parser.py [hunk 2]
```diff
@@ -357,5 +358,5 @@ def _write_parts_list(path, master_parts_list):
                  default_flow_style=False))
 
 
-if __name__ == '__main__':  # pragma: no cover
-    main()                  # pragma: no cover
+if __name__ == '__main__':        # pragma: no cover
+    sys.exit(1 if main() else 0)  # pragma: no cover
```
Comment (sergiusens) [inline snapcraft/internal/parser.py]: I think I liked my version better. This feels harder to read.

Comment (josepht) [inline snapcraft/internal/parser.py]: Fixed.

**Comment by elopio** [comment]
This wfm, but you say on the bug that you want to exit with 2 on origin errors.
If that's still the case, maybe you can open a different bug for that.

**Comment by josepht** [comment]
I'll do that.  I don't think it's a very high priority.

**Comment by elopio** [comment]
⟫ /home/elopio/workspace/canonical/snapcraft/bin/snapcraft-parser -i /tmp/test
Invalid wiki entry: Missing key in wiki entry: 'project-part'
1 wiki errors found!
⟫ echo $?
1

:shipit:

**Comment by sergiusens** [comment]
+1 waiting for adt to finish to merge

**Pull request merged**